### PR TITLE
ci: add docker build action

### PR DESCRIPTION
**Pull request by SOF3**

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          else
+            echo "tag=dev"
+          fi >>$GITHUB_OUTPUT
+
+      - name: Build docker image
+        env:
+          REGISTRY: ghcr.io/kubewharf
+          TAG: ${{ steps.tag.outputs.tag }}
+          ARCHS: amd64
+          OUTPUT_TYPE: registry
```
Comment (qclc) [inline .github/workflows/docker-build.yml:42]: Use the `registry` OUTPUT_TYPE, which is shorthand for `type=image,push=true`. But there is also github action that perform push below, and it seems that two push operations have been performed.

File: hack/make-rules/build-images.sh [hunk 2]
```diff
@@ -34,7 +34,7 @@ platform_array=("${arch_array[@]/#/linux/}")
 PLATFORMS=$(IFS=","; echo "${platform_array[*]}")
 REGISTRY=${REGISTRY:-"ghcr.io/kubewharf"}
 OUTPUT_TYPE="docker"
```
Comment (qclc) [inline hack/make-rules/build-images.sh:36]:
```suggestion
OUTPUT_TYPE=${OUTPUT_TYPE:-"docker"}
```
The value of `OUTPUT_TYPE` here is fixed as the `docker` type. If we need to support multiple types of `OUTPUT_TYPE`, please change it to receive a different variable value.

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (SOF3) [reply .github/workflows/docker-build.yml:42]: According to my previous test, `registry` only tags the image without pushing.

Comment (SOF3) [reply .github/workflows/docker-build.yml:42]: Oh it might be related to the comment below.

**Review by qclc — COMMENTED** [review]
lgtm

**Review by gary-lgy — APPROVED** [review]

**Pull request merged**